### PR TITLE
LPD-17073

### PR DESCRIPTION
**Pull request by daiazius**

### CODE REVIEW

**Comment by daiazius** [comment]
ci:test:sf

**Comment by daiazius** [comment]
ci:test:relevant

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/TicketResourceImpl.java [hunk 1]
```diff
@@ -45,32 +48,50 @@ public Ticket getUserAccountPasswordResetTicket(Long userAccountId)
...
-			_ticketLocalService.getTickets(
-				User.class.getName(), userAccountId, type);
+		PasswordPolicy passwordPolicy = user.getPasswordPolicy();
 
-		if (tickets.isEmpty()) {
-			return null;
+		Date expirationDate = null;
+
+		if ((passwordPolicy != null) &&
+			(passwordPolicy.getResetTicketMaxAge() > 0)) {
```
Comment (stian-sigvartsen) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/TicketResourceImpl.java:62]: If this condition is not met, then the ticket is created with no `expirationDate`. This means it remains valid forever, @daiazius can you review?

**Comment by daiazius** [comment]
ci:test:sf

**Comment by daiazius** [comment]
ci:test:relevant

**Comment by stian-sigvartsen** [comment]
Looks good @daiazius .
I rebased the PR reverted the changes to UserLocalService because existing methods could achieve the same thing there.

Please see liferay-appsec/liferay-portal/pull/1629

**Pull request closed without merging**